src/os/FileStore.cc

```diff
@@ -4228,10 +4270,17 @@ int FileStore::_setattrs(coll_t cid, const ghobject_t& oid, map<string,bufferptr
 	omap_set[p->first].push_back(p->second);
 	continue;
     }
-    omap_remove.insert(p->first);
-    inline_set.insert(*p);
-
-    inline_to_set.insert(*p);
+    if (spill_out)
+      omap_remove.insert(p->first);
+    if (inline_set.count(p->first) && inline_set[p->first] > 0) {
```

Comment:
@XinzeChi,  I suggest that we would use map<string, int>::iterator iter_inline = inline_set.find(p->first);

and do  iter_inline ==/!= inline_set.end() instead of inline_set.count(),  Since every inline_set.count or inline_set[] operation will search the map tree, while a single inline_set.find() operation just search the map tree once. You may try it.